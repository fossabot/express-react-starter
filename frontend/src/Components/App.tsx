import * as React from "react";

interface IAppProps {

}

interface IAppState {

}

/**
 * Standard app class that wraps everything
 */
class App extends React.Component<IAppProps, IAppState> {
    public render() {
        return "Hello world";
    }
}

export default App;