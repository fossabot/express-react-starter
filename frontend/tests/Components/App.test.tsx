import * as React from "react";
import App from "Components/App";
import { shallow } from "enzyme";

describe("Test App Component", () => {
    const app = shallow(<App />);

    it("Renders correctly", () => {
        expect(app).toMatchSnapshot();
    });

    it("Reacts to click", () => {
        app.find("div").simulate("click");

        expect(app).toMatchSnapshot();
    });
});