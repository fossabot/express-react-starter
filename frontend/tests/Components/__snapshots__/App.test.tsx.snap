// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test App Component Reacts to click 1`] = `
<Provider>
  <IntlProvider
    key="en"
    locale="en"
  >
    <FormattedMessage
      defaultMessage="Hello world"
      id="sidia.hello_world"
      values={Object {}}
    />
    <div
      onClick={[Function]}
    >
      asdf
    </div>
  </IntlProvider>
</Provider>
`;

exports[`Test App Component Renders correctly 1`] = `
<Provider>
  <IntlProvider
    key="en"
    locale="en"
  >
    <FormattedMessage
      defaultMessage="Hello world"
      id="sidia.hello_world"
      values={Object {}}
    />
    <div
      onClick={[Function]}
    >
      123
    </div>
  </IntlProvider>
</Provider>
`;
